Tests/SwiftDocCTests/Semantics/MetadataTests.swift

```diff
@@ -433,5 +433,32 @@ class MetadataTests: XCTestCase {
         
         return (problemIDs, metadata)
     }
+
+    func testInvalidMetadataDirectivesInDocumentationCommentHaveSolution() throws {
+        let source = """
+        @Metadata {
+            @DisplayName("Custom Name")
+            @TechnologyRoot
+        }
+        """
+        let document = Document(parsing: source, options: .parseBlockDirectives)
+        let directive = document.child(at: 0)! as! BlockDirective
+        let (bundle, _) = try testBundleAndContext()
+        var problems = [Problem]()
+        let metadata = Metadata(from: directive, source: nil, for: bundle, problems: &problems)
+        
+        metadata?.validateForUseInDocumentationComment(symbolSource: nil, problems: &problems)
+        
+        XCTAssertEqual(problems.count, 2)
+        
+        // Verify that each problem has a solution to remove the directive
+        for problem in problems {
+            XCTAssertNotNil(problem.possibleSolutions)
+            XCTAssertEqual(problem.possibleSolutions?.count, 1)
+            XCTAssertEqual(problem.possibleSolutions?.first?.summary, "Remove this \(problem.diagnostic.identifier.split(separator: ".").last!) directive.")
```

Comment:
I think that this test assertion will fail because the string it's expecting doesn't put single quotes around the directive name like the solution does.

FYI: A completely different style that some will find more readable and some will find less readable could be to move the summary assertion out of the for-loop and assert on both messages together:

```
XCTAssertEqual(problems.map(\.possibleSolutions.first?.summary), [
    "Remove this 'DisplayName' directive.",
    "Remove this 'TechnologyRoot' directive.",
])
```

(I believe that the order of the problems will be stable based on the order that they appear in the test content. Otherwise you can unwrap the first solution and sort the mapped summaries.)